viz_scripts/timeseries_analysis.ipynb

```diff
@@ -0,0 +1,471 @@
+    "quality_text = scaffolding.get_quality_text(participant_ct_df, expanded_ct)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate energy impact\n",
+    "expanded_ct = scaffolding.energy_intensity(expanded_ct, df_EI, 'Replaced_mode', 'Mode_confirm')\n",
+    "expanded_ct = scaffolding.energy_impact_kWH(expanded_ct, 'distance_miles', 'Replaced_mode', 'Mode_confirm')\n",
+    "expanded_ct = scaffolding.CO2_impact_lb(expanded_ct, 'distance_miles', 'Replaced_mode', 'Mode_confirm')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data Preprocessing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get timestamp from known year/month/day aggregated to days\n",
+    "data = expanded_ct[['user_id','start_local_dt_year','start_local_dt_month','start_local_dt_day','Mode_confirm','Mode_confirm_EI(kWH)','Mode_confirm_lb_CO2','distance_miles']].copy()\n",
+    "data.rename(columns={'start_local_dt_year':'year','start_local_dt_month':'month','start_local_dt_day':'day'}, inplace=True)\n",
+    "data['date_time'] = pd.to_datetime(data[['year','month','day']])\n",
+    "data = data.drop(columns=['year','month','day'])\n",
+    "\n",
+    "# Categorical type will include all days/modes in groupby even if there is no data for a particular tabulation\n",
+    "data.user_id = pd.Categorical(data.user_id)\n",
+    "data.date_time = pd.Categorical(data.date_time)\n",
+    "data.Mode_confirm = pd.Categorical(data.Mode_confirm,\n",
+    "                                   ordered=True,\n",
+    "                                   categories=['Car, drove alone',\n",
+    "                                              'Car, with others',\n",
+    "                                              'Taxi/Uber/Lyft',\n",
+    "                                              'Bus',\n",
+    "                                              'Free Shuttle',\n",
+    "                                              'Train',\n",
+    "                                              'Bikeshare',\n",
+    "                                              'Pilot ebike',\n",
+    "                                              'Regular Bike',\n",
+    "                                              'Scooter share',\n",
+    "                                              'Skate board',\n",
+    "                                              'Walk',\n",
+    "                                              'Other',\n",
+    "                                              'Not a Trip'])\n",
+    "\n",
+    "data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Count the number of trips for each sensed mode\n",
```

Comment:
nit: `confirmed mode`, not `sensed mode` in the comment

Reply:
Switched to confirmed mode